### PR TITLE
Feature/user interface

### PR DESCRIPTION
**Pull request by maosan132**
- Make some initial game logic
- All inside new feature branch "user_interface"

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi @maosan132  👋

You've done quite well on this, however, kindly make the following changes:

- [ ] We only want to break the lop when there is a win or a tie otherwise the loop needs to keep running, but yours is set to run only nine(9) times, this shouldn't be.

 - [ ] In addition within the body of the loop we want to display the board after each turn, yours doesn't display the board within each turn.

- [ ] Also within the body of the loop we also want to inform the player who just made a move if the move if a winning move or a tie, so this means you have to input a line perhaps as an inline comment where you check if the move just made is a winning move or a tie move.

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

File: bin/main.rb [hunk 1]
```diff
@@ -1,3 +1,75 @@
...
+puts '         ' + board[0].join(' | ')
+puts '        ———┼———┼———'
+puts '         ' + board[1].join(' | ')
+puts '        ———┼———┼———'
+puts '         ' + board[2].join(' | ')
+puts
+# ends board class
+counter = 0
+move_request = 'Select a numpad key to make your move'
+0.upto(9) do
```
Comment (mikenath223) [inline bin/main.rb]: Kindly change this line and exit only when the game is over, wither when there is a draw or a winner.

File: bin/main.rb [hunk 2]
```diff
@@ -1,3 +1,75 @@
...
+  move = gets.chomp
+  if counter.even?
+    puts "#{first_player}, you selected " + move
+  else
+    puts "#{second_player}, you selected " + move
+  end
+  puts
+  counter += 1
+  if counter < 9
+    puts 'Next player, make a move!'
```
Comment (mikenath223) [inline bin/main.rb]: Somewhere at this point, we want to display the board. So the next player can see the filled positions on the board.

**Review by serdg0 — CHANGES_REQUESTED** [review]
Hello @maosan132 !

You are doing a great job! :racing_car: 
The loop and the display board is awesome. Congratulations!
There are some implementation details we need to fix before moving forward.

- [ ] It's not a good practice to control your main loop with numbers. (0..8)

In other words, make the exit condition for the loop to be when there is a winner or a tie. This is because controling it with numbers can lead to bugs at the long run, in future milestones.

**Happy Coding!** :computer: :nerd_face:
**_Sergio Diaz_** (edited)

**Review by kobiyoyo — APPROVED** [review]
# Project Approved :thumbsup:
## Hello Team ,  You have done a great job with this project. :clap:
#### You can go ahead and merge !!!

**Pull request merged**